BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -540,13 +540,15 @@ - (void)processResponse:(NSURLResponse *)response
 
 - (void)finishURLSessionTaskWithData:(NSData *)data response:(NSURLResponse *)response error:(NSError *)error
 {
-    if (self.HTTPResponse == nil) {
+    if (self.HTTPResponse == nil && response != nil) {
```

Comment:
If we call processResponse with a nil response, it will always generate an error. If there is no response to process, we should not go through that - particularly here on 'finish' on a 'processIntermediateResponse' a nil response SHOULD generate an error but in this case it can be valid (for example a network error where the failure happens before reaching the server, like no connection or a DNS problem.)